mhctools/process_helpers.py

```diff
@@ -108,8 +112,10 @@ def run_multiple_commands_redirect_stdout(
 
     def add_to_queue(process):
         if print_commands:
-            print(" ".join(process.args), ">",
-                  process.redirect_stdout_file.name)
+            handler = logging.FileHandler(process.redirect_stdout_file.name)
+            handler.setLevel(logging.DEBUG)
+            logger.addHandler(handler)
```

Comment:
so in that situation, does it currently write to tens of thousands of files? or it runs 20k times but only writes to a handful of files?